app/src/main/java/com/github/sdp/mediato/data/CollectionsDatabase.java

```diff
@@ -0,0 +1,89 @@
+package com.github.sdp.mediato.data;
+
+import com.github.sdp.mediato.model.Review;
+import com.github.sdp.mediato.model.media.Collection;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.concurrent.CompletableFuture;
+
+public class CollectionsDatabase {
+    private static final String USERS_PATH = "Users/";
+    private static final String REVIEWS_PATH = "reviews/";
+    private static final String USER_COLLECTIONS_PATH = "/collections/";
+
+    public static FirebaseDatabase database = FirebaseDatabase.getInstance();
+
+    /**
+     * Adds a collection to the user
+     *
+     * @param username   the concerned user
+     * @param collection the collection to be added
+     */
+    public static void addCollection(String username, Collection collection) {
+        getCollectionReference(username, collection.getCollectionName())
+                .setValue(collection)
+                .addOnCompleteListener(task -> System.out.println("Added " + collection.getCollectionName() + " to " + username));
+    }
+
+    /**
+     * Remove a collection from a user
+     *
+     * @param username
+     * @param collectionName
+     */
+    public static void removeCollection(String username, String collectionName) {
+        getCollectionReference(username, collectionName)
+                .setValue(null)
+                .addOnCompleteListener(task -> System.out.println("Removed " + collectionName + " from " + username));
+    }
+
+    /**
+     * Retrieves a collection
+     *
+     * @param username
+     * @param collectionName
+     * @return
+     */
+    public static CompletableFuture<Collection> getCollection(String username, String collectionName) {
+        CompletableFuture<Collection> future = new CompletableFuture<>();

```

Comment:
Just a question: what happens if there would be two collections of the same name for a user? Would it just return the first it finds or is creating two collections of the same name in the database not possible? It should not be a problem though since I disallowed using the same collection name in the UI.